packages/explat-client/src/internal/timing.ts

```diff
@@ -0,0 +1,57 @@
+export const MILLISECONDS_PER_SECOND = 1000;
+
+let lastNow = Date.now();
+/**
+ * Returns the time in miliseconds.
+ * A strictly increasing version of Date.now()
+ */
+export function monotonicNow(): number {
+	const maybeNow = Date.now();
+
+	lastNow = lastNow < maybeNow ? maybeNow : lastNow + 1;
+
+	return lastNow;
+}
+
+export function createUnresolvingPromise< T >(): Promise< T > {
+	return new Promise< T >( () => {
+		return;
+	} );
+}
+
+/**
+ * Timeouts a promise. Returns timeoutValue in event of timeout.
+ *
+ * @param promise The promise to timeout
+ * @param ms The timeout time in milliseconds
+ */
+export function timeoutPromise< T >( promise: Promise< T >, ms: number ): Promise< T | null > {
+	return Promise.race( [
+		promise,
+		new Promise< null >( ( res ) => setTimeout( () => res( null ), ms ) ),
```

Comment:
What do you think of throwing an exception on timeout instead of returning null?

Reply:
Yeah I'm fine with that.